### PR TITLE
OneBranch Migration

### PR DESCRIPTION
**Pull request by trwalke**
# OneBranch Migration

- [ ] You've read the [Contributor Guide](https://github.com/AzureAD/microsoft-identity-web/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/AzureAD/microsoft-identity-web/blob/master/CODE_OF_CONDUCT.md).
- [ ] You've included unit or integration tests for your change, where applicable.
- [ ] You've included inline docs for your change, where applicable.
- [ ] There's an open issue for the PR that you are making. If you'd like to propose a new feature or change, please open an issue to discuss the change or find an existing issue.

Summary of the changes (Less than 80 chars)

## Description

Updating YAML to enable OneBranch Build.
SDL tasks such as credscan, policheck, tsa upload, roslyn/codeQL are handled by OneBranch auto injected tasks.

This build only produces signed packages. Tests will come when MSI is working on the pipeline.

Working Legacy build: https://identitydivision.visualstudio.com/IDDP/_build/results?buildId=1287509&view=results
Working OneBranch Build: https://dev.azure.com/IdentityDivision/IDDP/_build/results?buildId=1288785&view=results
Fixes #{bug number} (in this specific format)

IdWebSourceDir is a variable set on the Legacy and OneBranch builds to specify the source folder.
PipelineType is a variable set on the Legacy and OneBranch builds to specify the type of pipeline. Used to disable tasks on Onebranch

Onebranch PR on ADO: https://dev.azure.com/IdentityDivision/IDDP/_git/Microsoft%20Identity%20Web%20-%20OneBranch/pullrequest/12564?path=/owners.txt

### CODE REVIEW

File: build/template-onebranch-release-build.yaml [hunk 1]
```diff
@@ -0,0 +1,24 @@
+# template-onebranch-release-build.yaml
+
+parameters:
+  BuildPlatform: 'any cpu'
+  BuildConfiguration: 'debug'
+  MsIdentityWebSemVer: '1.0.0-devopsbuild'
```
Comment (bgavrilMS) [inline build/template-onebranch-release-build.yaml:6]: I recommend something like `1.0.0-internal$(Build.BuildNumber)`

**Review by bgavrilMS — APPROVED** [review]

**Comment by jennyf19** [comment]
where do the packages go now? we check them manually to ensure things are signed correctly.

**Comment by jennyf19** [comment]
there is an npm error in ADO:
image

and some warnings seem relevant to these changes and should be fixed.

**Review by jennyf19 — CHANGES_REQUESTED** [review]
left a few questions

**Comment by trwalke** [comment]
Looking at warnings/error.

Packages can be found here @jennyf19 
image

**Comment by trwalke** [comment]
> there is an npm error in ADO: image
> 
> and some warnings seem relevant to these changes and should be fixed.

@jennyf19 resolved the error that you were seeing in the latest build.
A lot of the additional warnings are a result of the Onebranch tasks and I dont really have any way to resolve them. 
Most of them can be ignored as they do not affect the operation of the build.
for example: 
image

File: build/template-onebranch-release-build.yaml [hunk 2]
```diff
@@ -0,0 +1,24 @@
+# template-onebranch-release-build.yaml
+
+parameters:
+  BuildPlatform: 'any cpu'
+  BuildConfiguration: 'debug'
```
[the remark this answers is not in the document]
Comment (trwalke) [reply build/template-onebranch-release-build.yaml:5]: yes

**Review by FuPingFranco — APPROVED** [review]
LGTM, Thanks @trwalke!

**Comment by FuPingFranco** [comment]
@trwalke Could you update the official release guidelines https://identitydivision.visualstudio.com/DevEx/_git/MISE?path=/docs/releases/OfficialReleases.md&version=GBmaster&_a=previewonce both PR's are in. Thanks!

**Pull request merged**